astroplan/moon.py

```diff
@@ -0,0 +1,170 @@
+        Position of the moon at ``time``
+    """
+    if not isinstance(time, Time):
+        time = Time(time)
+
+    if use_pyephem:
+        try:
+            import ephem
+        except ImportError:
+            raise ImportError("The get_moon function currently requires "
+                              "PyEphem to compute the position of the moon.")
+
+        moon = ephem.Moon()
+        obs = ephem.Observer()
+        obs.lat = location.latitude.to(u.degree).to_string(sep=':')
+        obs.lon = location.longitude.to(u.degree).to_string(sep=':')
+        obs.elevation = location.height.to(u.m).value
+        if pressure is not None:
+            obs.pressure = pressure.to(u.bar).value*1000.0
+
+        if time.isscalar:
+            obs.date = time.datetime
+            moon.compute(obs)
+            moon_ra = float(moon.ra)
+            moon_dec = float(moon.dec)
+            moon_dist = moon.earth_distance
+        else:
+            moon_ra = []
+            moon_dec = []
+            moon_dist = []
+            for t in time:
+                obs.date = t.datetime
+                moon.compute(obs)
+                moon_ra.append(float(moon.ra))
+                moon_dec.append(float(moon.dec))
+                moon_dist.append(moon.earth_distance)
+
+    else:
+        try:
+            import jplephem
+        except ImportError:
+            raise ImportError("The get_moon function currently requires "
+                              "jplephem to compute the position of the moon.")
+
+        # Calculate position of moon relative to Earth by subtracting the
+        # vector pointing from the Earth-moon barycenter to the moon by
+        # the vector from the Earth-moon barycenter to the Earth
+        from jplephem.spk import SPK
+        kernel = SPK.open(get_spk_file())
+        cartesian_position = (kernel[3,301].compute(time.jd) -
+                              kernel[3,399].compute(time.jd))
+        x, y, z = cartesian_position*u.km
+
+        # Convert to GCRS coordinates
+        cartrep = CartesianRepresentation(x=x, y=y, z=z)
+        return SkyCoord(cartrep, frame=GCRS(obstime=time,
+                                            obsgeoloc=location))
+
+    moon_sc = SkyCoord(ra=Longitude(moon_ra, unit=u.rad),
+                       dec=Latitude(moon_dec, unit=u.rad),
+                       distance=u.Quantity(moon_dist, u.AU), frame='gcrs')
+
+    return moon_sc
+
+def calc_moon_phase_angle(time, location):
+    """
+    Calculate lunar orbital phase [radians].
+
+    Parameters
+    ----------
+    moon : `~astropy.coordinates.SkyCoord`
+        Position of the moon
+
+    sun : `~astropy.coordinates.SkyCoord`
+        Position of the Sun
+
+    Returns
+    -------
+    i : float
+        Phase angle of the moon [radians]
+    """
+    # TODO: cache these sun/moon SkyCoord objects
+    sun = get_sun(time)
+    #sun = sun.transform_to(AltAz(location=location, obstime=time))
```

Comment:
@eteq : When I do this line to transform to `AltAz` as you [suggested](https://github.com/astroplanners/astroplan/pull/39#issuecomment-126506485), I get an infinite recursion, so I've commented it out. Am I doing something silly here?